.github/workflows/qemu-start.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/python3
+
+import pexpect
+import sys
+import time
+import argparse
+
+
+def start_qemu(target, subtarget):
+    qemu_cmd = f"./scripts/qemustart {target} {subtarget} --netdev user,id=wan -netdev user,id=lan,net=192.168.1.0/24,dhcpstart=192.168.1.100,restrict=yes,hostfwd=tcp::8022-:22 -net nic,netdev=wan -net nic,netdev=lan"
+    print(f'calling: "{qemu_cmd}"')
+    child = pexpect.spawn(qemu_cmd, logfile=sys.stdout.buffer)
+
+    child.expect("Please press Enter to activate this console.", timeout=80)
+    child.expect("link becomes ready", timeout=90)
+    child.send("\n")
+
+    child.expect(":/#", timeout=10)
+    child.sendline("uname -a")
+    # Wait till network is really available
+    time.sleep(4)
+    return child
+
+
+def configure_network(child):
+    child.expect(":/#")
+    child.sendline("ip addr")
+    child.expect("192.168.1.1/24")
+
+    child.expect(":/#")
+    child.sendline("uci set network.lan.proto=dhcp")
+
+    child.expect(":/#")
+    child.sendline("uci changes")
+    child.expect("network.lan.proto='dhcp'")
+
+    child.expect(":/#")
+    child.sendline("uci commit")
+
+    child.expect(":/#")
+    child.sendline("service network reload")
+    child.expect("entered forwarding state")
+    # Wait for DHCP configuration
+    time.sleep(8)
+
+    child.expect(":/#")
+    child.sendline("ip addr")
+    child.expect("10.0.2.15/24")
+    return child
+
+
+def check_https_download(child, url, expect_str, expect_code=0):
+    child.expect(":/#")
+    child.sendline(f"wget {url}")
+    child.expect(expect_str)
+
+    child.expect(":/#")
+    child.sendline("echo $?")
+    child.expect(str(expect_code))
+    return child
+
+
+def check_download(child):
+    child = check_https_download(
+        child,
+        "https://downloads.openwrt.org/releases/21.02.2/targets/armvirt/64/config.buildinfo",
+        "Download completed ",
+    )
+
+    child.expect(":/#")
+    child.sendline("sha256sum config.buildinfo")
+    child.expect(
+        "26b85383a138594b1197e581bd13c6825c0b6b5f23829870a6dbc5d37ccf6cd8  config.buildinfo"
+    )
+
+    child = check_https_download(
+        child,
+        "https://untrusted-root.badssl.com/",
+        "Connection error: Invalid SSL certificate",
+        5,
+    )
+
+    child = check_https_download(
+        child,
+        "https://wrong.host.badssl.com/",
+        "Connection error: Server hostname does not match SSL certificate",
+        5,
+    )
+
+    return child
+
+
+def check_ssh():
+    ssh = pexpect.spawn(
+        'ssh-keygen -f "/home/hauke/.ssh/known_hosts" -R "[localhost]:8022"',
```

Comment:
Is this hardcoded user name correct?

Reply:
sorry I should fix this. We should probably use some parameters to the ssh command to ignore the server key.

Reply:
It is fixed now, I changed the ssh command to `ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null root@localhost -p 8022` then it ignores the current key.